idx/memory/find_cache.go

```diff
@@ -69,22 +96,26 @@ func (c *FindCache) Get(orgId uint32, pattern string) ([]*Node, bool) {
 func (c *FindCache) Add(orgId uint32, pattern string, nodes []*Node) {
 	c.RLock()
 	cache, ok := c.cache[orgId]
-	t := c.backoff[orgId]
+	backoff := findCacheBackoff.Peek()
```

Comment:
I like this approach, but findCacheBackoff cant be a global, it needs to be part of the FindCache struct.

Reply:
actually, findCacheBackoff metric doesnt work at all with the partitioned Index, as there are multiple findCaches that could all have different states.

Reply:
why, to make it clear that no other code in the package should mess with it? if that, we could solve it with a better comment

Reply:
I would just add a "backoff" field to the FindCache struct.   Then in triggerBackoff() i would
```
time.AfterFunc(c.backoffTime, func() {
   findCacheBackoff.SetFalse
   c.Lock()
   c.backoff = false
   c.Unlock()
})
```

Reply:
> why, to make it clear that no other code in the package should mess with it

It is just a bad idea to mix structs and fields together.   you can have multiple instances of a findCache at any one time, so controlling how the instance behaves through a global variable is just a recipe for disaster.

Reply:
> actually, findCacheBackoff metric doesnt work at all with the partitioned Index, as there are multiple findCaches that could all have different states.

 hmm I didn't consider this. good catch. will follow your suggestion

Reply:
actually to accommodate PartitionedIndex, we can't use a backoff bool either. maybe it should be a gauge that is incremented and decremented.